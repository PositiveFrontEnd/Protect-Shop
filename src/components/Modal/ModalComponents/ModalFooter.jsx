import React from "react";
import Button from "../../Button/Button";
import PropTypes from 'prop-types'

const ModalFooter = ({ className, firstText, firstClassName, secondaryClassName,secondaryText,firstClick,secondaryClick})=>{
    return(
        <div className={className}>
            {secondaryText && <Button className={secondaryClassName} white click={secondaryClick}>{secondaryText}</Button>}
         {firstText && <Button className={firstClassName} black click={firstClick}>{firstText}</Button>}

        </div>
    )
}

ModalFooter.propTypes = {
    className: PropTypes.string,
    firstText: PropTypes.string,
    secondaryText: PropTypes.string,
    firstClick: PropTypes.func,
    secondaryClick: PropTypes.func,
    firstClassName: PropTypes.string,
    secondaryClassName: PropTypes.string
}

export default ModalFooter