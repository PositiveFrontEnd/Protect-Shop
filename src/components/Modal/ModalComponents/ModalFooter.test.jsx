import React from "react";
import { fireEvent, getByText, render, screen } from "@testing-library/react";
import ModalFooter from "./ModalFooter";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
describe("test Modal", () => {
    test("render Modal", () => {
        const mockStore = configureStore();
        const initialState = {
        }
        const data = {
            className: "some__className",
            firstText: "text one",
            firstClassName: "className__first",
            secondaryClassName: "className__secondary",
            secondaryText: "text secondary",
            firstClick: () => console.log("hello first"),
            secondaryClick: () => console.log("hello secondary")
        }

        const store = mockStore(initialState);
        const { asFragment } = render(
            <Provider store={store}>
                <BrowserRouter>
                    <ModalFooter data={data} />
                </BrowserRouter>
            </Provider>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});