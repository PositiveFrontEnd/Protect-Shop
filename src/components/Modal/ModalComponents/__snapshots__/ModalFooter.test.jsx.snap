// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test Modal > render Modal 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;
